internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,308 @@
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				props := model.Properties
+
+				state := CustomLocationResourceModel{
+					Name:                id.CustomLocationName,
+					ResourceGroupName:   id.ResourceGroupName,
+					Location:            model.Location,
+					ClusterExtensionIds: pointer.From(props.ClusterExtensionIds),
+					DisplayName:         pointer.From(props.DisplayName),
+					HostResourceId:      pointer.From(props.HostResourceId),
+					HostType:            string(pointer.From(props.HostType)),
+					Namespace:           pointer.From(props.Namespace),
+				}
+
+				if props.Authentication != nil && props.Authentication.Type != nil && props.Authentication.Value != nil {
+					state.Authentication = []AuthModel{
+						{
+							Type:  pointer.From(props.Authentication.Type),
+							Value: pointer.From(props.Authentication.Value),
+						},
+					}
+				}
+
+				return metadata.Encode(&state)
+			}
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state CustomLocationResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{}
+			d := metadata.ResourceData
+
+			if d.HasChanges("authentication") {
+				if len(state.Authentication) > 0 {
+					auth := state.Authentication[0]
+					customLocationProps.Authentication = &customlocations.CustomLocationPropertiesAuthentication{
+						Type:  pointer.To(auth.Type),
+						Value: pointer.To(auth.Value),
+					}
+				} else {
+					customLocationProps.Authentication = nil
```

Comment:
We're using a `PATCH` - Is `nil` here just a no-op as would be expected, or does it actually remove the value? Can we add test coverage that shows it's successfully removed?